### DISCUSSION TITLE
Server side user auth

### DISCUSSION THREAD

**Discussion by abjelosevic88** [supabase/supabase#1874] [Questions] [answered]
I am trying to make queries over a server and I am unable to pass user access_token properly.

I am using Next JS and this a code that I am using:

`supabase.createClient(
    process.env.NEXT_PUBLIC_SUPABASE_URL,
    process.env.NEXT_PUBLIC_SUPABASE_KEY,
    {
      headers: {
        Authorization: 'Bearer ' + token
      }
    }
  )`

Later on in the code I am trying to update blog object:
`const { data: blogs, error } = await supabase.from('blogs').update(updateObj).eq('id', data.id)`

Response from server:
`blogs null, error []`

Everything was set up correctly current user is the one that 'owns' this blog item.

What am I doing wrong here?

Thank you in advance

**Comment by easy-street** [comment, marked as answer]
the new `setAuth` method ([not yet documented](https://github.com/supabase/supabase/issues/1818)) supports this use case:
```
import { createClient } from '@supabase/supabase-js'

export const supabase = createClient(
  process.env.NEXT_PUBLIC_SUPABASE_URL,
  process.env.NEXT_PUBLIC_SUPABASE_KEY
)
```

then:
```
// `supabase` is the reference to the instantiated client
supabase.auth.setAuth(token)

const { data: blogs, error } = await supabase.from('blogs').update(updateObj).eq('id', data.id)
```

**Reply by abjelosevic88** [reply]
Yeah. That's it! Thank you!
